### PR TITLE
Update environment setup guide in README.md

### PR DESCRIPTION
**Pull request by pandamicro**
Re: #

### Changelog

* Update environment setup guide in README.md

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [x] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,22 +64,24 @@ Clone this repository into your local environment.
 In the cloned engine folder, run the following command to setup development environment:
 
 ```bash
-# download & build engine dependencies
+# download external dependencies
+cd engine-native
```
Comment (PPpro) [inline README.md]:
```suggestion
cd ./native
```

File: README.md [hunk 2]
```diff
@@ -64,22 +64,24 @@ Clone this repository into your local environment.
...
 npm install
+gulp init
+# back to the root folder
+cd ..
+# setup environment and build engine files
+npm install
+npm build
```
Comment (PPpro) [inline README.md]:
```suggestion
```

we don't need `npm run build` in `develop` branch anymore

**Review by PPpro — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -64,22 +64,23 @@ Clone this repository into your local environment.
...
-- If running inside Cocos Creator, the engine will automatically compile and build after the editor window is opened. For more instructions on modifying the engine in Cocos Creator, please refer to [Engine Customization Workflow](https://docs.cocos.com/creator/manual/en/advanced-topics/engine-customization.html).
-- Outside the editor, you need to run the following command to build:
-
-```bash
-npm run build
-```
+The Cocos Creator engine is designed to work with its editor, the engine will automatically compile and build after the editor window is opened. For more instructions on modifying the engine in Cocos Creator, please refer to [Engine Customization Workflow](https://docs.cocos.com/creator/manual/en/advanced-topics/engine-customization.html).
 
-Please refer to [native readme](native/README.md) if you want to develop native applications.
+Please refer to [native readme](native/README.md) for more information related to native development enviroment.
```
Comment (minggo) [inline README.md]: enviroment -> environment

Comment (pandamicro) [reply README.md]: fixed

**Review by minggo — APPROVED** [review]

**Pull request merged**